scripts/generate/templates/TypedSlot.js

```diff
@@ -0,0 +1,32 @@
+const format = require('../format-lines');
+const { TYPES } = require('./Slot.opts');
+
+const header = `\
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Library for representing bytes32 slot as typed objects.
+ */
+`;
+
+const udvt = ({ type, name }) => `\
+/**
+ * @dev UDVT that represent a slot holding a ${type}.
+ */
+type ${name}SlotType is bytes32;
+
+/**
+ * @dev Cast an arbitrary slot to a ${name}SlotType.
+ */
+function as${name}Slot(bytes32 slot) internal pure returns (${name}SlotType) {
```

Comment:
I'm not sure. I think "asUint256Slot" is more representative.

`toDelay` takes a duration that is a uint32, and transform it to a delay object, that is a uint112, and that represent more (there is space for the pending value, the effect timepoint...). So IMO its a "to" because there is a transformation (even if its just adding zeros)

In this case, its a bytes32 that remains a bytes32 ... just with a wrapper. Its really like `as` in typescript.